### PR TITLE
ENH: adding true positives, true negatives, false positives and false…

### PR DESCRIPTION
**Pull request by maxwell-aladago**
Adding true positives, true negatives, false positives, false negatives to metrics as well as correctness tests

### CODE REVIEW

**Comment by PhilipMay** [comment]
Hi @maxwell-aladago ,
thanks for this contribution.

For me it seems like you do a small mistake here:
To implement this you need a stateful metric. This metric here only works on batch level.

It is done by setting `self.stateful = True` on `keras.layers.Layer`.

Also have a look here: https://github.com/keras-team/keras/blob/master/tests/keras/metrics_test.py#L130

With an example implementation here: https://github.com/netrack/keras-metrics/blob/master/keras_metrics/metrics.py

Here is the keras code position where stateful is handled: https://github.com/keras-team/keras/blob/master/keras/engine/training.py#L440

**Comment by PhilipMay** [comment]
Disregarding the implementation on [keras_metrics](https://github.com/netrack/keras-metrics/blob/master/keras_metrics/metrics.py) I think it would be very useful to implement more metrics directly here in keras.

**Comment by maxwell-aladago** [comment]
> Hi @maxwell-aladago ,
> thanks for this contribution.
> 
> For me it seems like you do a small mistake here:
> To implement this you need a stateful metric. This metric here only works on batch level.
> 
> It is done by setting `self.stateful = True` on `keras.layers.Layer`.
> 
> Also have a look here: https://github.com/keras-team/keras/blob/master/tests/keras/metrics_test.py#L130
> 
> With an example implementation here: https://github.com/netrack/keras-metrics/blob/master/keras_metrics/metrics.py
> 
> Here is the keras code position where stateful is handled: https://github.com/keras-team/keras/blob/master/keras/engine/training.py#L440

Oohk, didn't see these before. Thanks so much. This is very helpful

**Comment by PhilipMay** [comment]
> Oohk, didn't see these before. Thanks so much. This is very helpful

Also see the comments here: #13034

**Comment by PhilipMay** [comment]
@maxwell-aladago Do you plan to continue to work on this PR or just use [keras_metrics](https://github.com/netrack/keras-metrics/blob/master/keras_metrics/metrics.py)?

**Comment by maxwell-aladago** [comment]
> @maxwell-aladago Do you plan to continue to work on this PR or just use [keras_metrics](https://github.com/netrack/keras-metrics/blob/master/keras_metrics/metrics.py)?

I think I will just use [keras-metrics](https://github.com/netrack/keras-metrics/blob/master/keras_metrics/metrics.py). Thank you

**Comment by PhilipMay** [comment]
This PR can be closed imo.

**Comment by maxwell-aladago** [comment]
> This PR can be closed IMO (see above).

Yes, thank you
